app/javascript/shared/components/emoji/EmojiInput.vue

```diff
@@ -44,14 +94,65 @@ export default {
   },
   data() {
     return {
-      selectedKey: 'Smileys & Emotion',
+      selectedKey: 'Search',
       emojis,
+      search: '',
     };
   },
+  computed: {
+    categories() {
+      return this.emojis.map(category => category);
+    },
+    filterEmojisByCategory() {
+      const selectedCategoryName = this.emojis.find(category =>
+        category.name === this.selectedKey ? category.name : null
+      );
+      return selectedCategoryName?.emojis;
+    },
+    filterAllEmojisBySearch() {
+      return this.emojis.map(category => {
+        const allEmojis = category.emojis.filter(emoji =>
+          emoji.name.includes(this.search)
+        );
+        return allEmojis.length > 0
+          ? { ...category, emojis: allEmojis }
+          : { ...category, emojis: [] };
+      });
+    },
+    hasNoSearch() {
+      return this.selectedKey !== 'Search' && this.search === '';
```

Comment:
Can we make a constant for search key `SEARCH_KEY`

Reply:
Done